syne_tune/tuner_callback.py

```diff
@@ -189,62 +189,95 @@ def _set_time_fields(self, result: Dict):
+                self.writer.add_scalar(self.target_metric, result[self.target_metric], self.iter)
 
             else:
                 opt = self.metric_sign * self.curr_best_value
-                self.writer.add_scalar(self._target_metric, opt, self._iter)
+                self.writer.add_scalar(self.target_metric, opt, self.iter)
 
             for key, value in self.curr_best_config.items():
-                self.writer.add_scalar(f'optimal_{key}', value, self._iter)
+                self.writer.add_scalar(f'optimal_{key}', value, self.iter)
 
         for metric in result:
-            if self.ignore_metrics is not None and metric not in self.ignore_metrics:
-                self.writer.add_scalar(metric, result[metric], self._iter)
+            if metric not in self.ignore_metrics:
+                self.writer.add_scalar(metric, result[metric], self.iter)
 
         for key, value in trial.config.items():
-            self.writer.add_scalar(key, value, self._iter)
+            self.writer.add_scalar(key, value, self.iter)
 
-        self.writer.add_scalar('runtime', result[ST_TUNER_TIME], self._iter)
+        self.writer.add_scalar('runtime', result[ST_TUNER_TIME], self.iter)
 
         self.trial_ids.add(trial.trial_id)
         self.writer.add_scalar('number_of_trials', len(self.trial_ids),
-                               self._iter, display_name='total number of trials')
+                               self.iter, display_name='total number of trials')
 
-        self._iter += 1
+        self.iter += 1
 
     def on_tuning_start(self, tuner):
 
-        output_path = os.path.join(tuner.tuner_path, 'tensorboard_output')
+        self.output_path = os.path.join(tuner.tuner_path, 'tensorboard_output')
 
         try:
             from tensorboardX import SummaryWriter
         except ImportError:
             logger.error('TensoboardX is not installed. You can install it via: pip install tensorboardX')
-        self.writer = SummaryWriter(output_path)
-        self._iter = 0
-        self._start_time_stamp = perf_counter()
+        self.writer = SummaryWriter(self.output_path)
+        self.iter = 0
+        self.start_time_stamp = perf_counter()
 
     def on_tuning_end(self):
         self.writer.close()
 
     def __getstate__(self):
-        # To avoid runtime errors because of the MultiProcessing Queues of TensorboardX,
-        # we don't serialize the callback
-        return None
+        state = {
+            "results": self.results,
+            "ignore_metrics": self.ignore_metrics,
+            "curr_best_value": self.curr_best_value,
+            "curr_best_config": self.curr_best_config,
+            "start_time_stamp": self.start_time_stamp,
+            "iter": self.iter,
+            "mode": self.mode,
+            "target_metric": self.target_metric,
+            "trial_ids": self.trial_ids,
+            "metric_sign": self.metric_sign,
+            'output_path': self.output_path
```

Comment:
It seems that the formatting is not working in your case.
Can you verify that you have black setup? Have you ran `./dev_setup.sh` to install black hook?